src/common/modules/PageHandler.js

```diff
@@ -35,6 +35,7 @@ export async function insertIntoPage(text) {
         });
 
         // send request to insert emoji
+        // This will not work in Manifest V3: https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#executing-arbitrary-strings
```

Comment:
Oh thanks, that's interesting. AFAIK that part of Manifest v3 [could be adopted by Firefox as well](https://blog.mozilla.org/addons/2022/05/18/manifest-v3-in-firefox-recap-next-steps/), but in any case, that can be re-written. It's anyway somewhat ugly to inject a new function call script snippet just to invoke a function. Likely a real messaging solution should be used here.

Reply:
Also, if that already fails in Chrome/ium, we should disable the corresponding option on the options page for that.

Reply:
Manifest V2 still works in Chrome until January 2023, although as of January 2022, the Chrome Web Store is no longer accepting new Manifest V2 submissions. See [here](https://developer.chrome.com/docs/extensions/mv3/mv2-sunset/) for more information.

Reply:
So that means we basically need to adjust it?

Reply:
Yes, this and all your extensions will need to be converted to Manifest V3 (see #68), but Firefox does not yet fully support MV3. According to the link you posted above, they "expect to launch MV3 support for all users by the end of 2022". Thunderbird likely will not fully support MV3 until the next ESR release in summer 2023 at the earliest (they still have not fully transitioned to web extensions).

Unless you would like to publish the extensions to the Chrome Web Store sooner, it probably would be best to wait until at least Firefox supports MV3 before switching. Besides this change, the background page will also need rather extensive changes to convert it to either an event page (Firefox only) or a service worker (Firefox and Chrome). IMO, upgrading to the latest version of Emoji Mart and fixing #116 would be a higher priority.